src/eval-frame/actions/actions.js

```diff
@@ -87,19 +89,20 @@ export function runCodeWithLanguage(language, code) {
 export async function evaluateCode(code, language, chunkId, evalId) {
   try {
     MOST_RECENT_CHUNK_ID.set(chunkId);
-    const output = await runCodeWithLanguage(language, code);
-    addConsoleEntryInEditor({
-      historyType: "CONSOLE_OUTPUT",
-      value: output
+    const value = await runCodeWithLanguage(language, code);
+
+    const historyId = addConsoleEntryInEditor({
+      historyType: "CONSOLE_OUTPUT"
     });
+    IODIDE_EVALUATION_RESULTS[historyId] = value;
```

Comment:
adding to IODIDE_EVALUATION_RESULTS now only happens in the case of true evaluation results, so we can do it here rather than when we add to things to the history